worker/utils/fastlyJob.js

```diff
@@ -1,92 +1,128 @@
 const request = require('request');
-const utils = require('../utils/utils');
 const environment = require('../utils/environment').EnvironmentClass;
 const fastly = require('fastly')(environment.getFastlyToken());
+const utils = require('../utils/utils');
+
+const fastlyServiceId = environment.getFastlyServiceId();
+const headers = {
+    'Fastly-Key': environment.getFastlyToken(),
+    'Accept': 'application/json',
+    'Content-Type': 'application/json',
+    'Fastly-Debug': 1,
+};
+
 
 class FastlyJobClass {
-  // pass in a job payload to setup class
-  constructor(currentJob) {
-    this.currentJob = currentJob;
-    if (fastly === undefined) {
-      utils.logInMongo(currentJob, 'fastly connectivity not found');
+    // pass in a job payload to setup class
+    constructor(currentJob) {
+        this.currentJob = currentJob;
+        if (fastly === undefined) {
+          utils.logInMongo(currentJob, 'fastly connectivity not found');
+        }
     }
-  }
 
-  // takes in an array of urls and purges cache for each
-  async purgeCache(urlArray) {
-    if (!Array.isArray(urlArray)) {
-      throw new Error('Parameter `urlArray` needs to be an array of urls');
-    }
+    // takes in an array of surrogate keys and purges cache for each
+    async purgeCache(urlArray) {
+        if (!Array.isArray(urlArray)) {
+            throw new Error('Parameter `urlArray` needs to be an array of urls');
+        }
 
-    let that = this;
-    let urlCounter = urlArray.length;
-    let purgeMessages = [];
+        for (let i = 0; i < urlArray.length; i++) {
+              // retrieve surrogate key
+              try {
+                  const surrogateKey = await this.retrieveSurrogateKey(urlArray[i]);
+                  // perform request to purge
+                  await this.requestPurgeOfSurrogateKey(surrogateKey);
 
-    // the 1 is just "some" value needed for this header: https://docs.fastly.com/en/guides/soft-purges
-    const headers = {
-      'fastly-key': environment.getFastlyToken(),
-      'accept': 'application/json',
-      'Fastly-Soft-Purge': '1',
-    };
+                  // warm cache
+                  await this.warmCache(urlArray[i]);
+              } catch (error) {
+                  console.trace(error);
+                  // should return reject here?
```

Comment:
same issue here, do we need to be issuing rejects where we issue resolves? ie do we need to have a condition for rejecting? or is catching and throwing an error enough?